ambry-commons/src/main/java/com/github/ambry/compression/LZ4Compression.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.compression;
+
+import net.jpountz.lz4.LZ4Compressor;
+import net.jpountz.lz4.LZ4Factory;
+import net.jpountz.lz4.LZ4FastDecompressor;
+
+/**
+ * The LZ4 compression algorithm.  LZ4 has very fast decompression speed.
+ * More info about LZ4  <a href="https://github.com/lz4/lz4">GitHub</a>.
+ *
+ * LZ4 has 3 implementations and LZ4Factory.fastestInstance() picks the implementation in this order
+ * 1. Native with JNI.
+ * 2. Pure Java unsafe instance
+ * 3. Pure Java safe instance
+ *
+ * Each LZ4 compressor offers 2 compressors.
+ * - LZ4 fast compressor (uses about 16 KB memory) for fastest compression speed but lower compression ratio.
+ *   To use fast compressor, set compression level to 0.
+ * - LZ4 high compressor (HC) (uses about 256 KB memory) for higher compression ratio but about 10x slower.
+ *   To use LZ4 HC, select a compression level between 1 and 17.  Higher compression level yields higher compression
+ *   ratio but slower compression speed.
+ *
+ * The decompressor can decompress data compressed in either compressors and the decompression speed is about the
+ * same regardless of which compression level was used.
+ */
+public class LZ4Compression extends BaseCompressionWithLevel {
+
+  // For performance purpose, cache the factory.
+  private static final LZ4Factory LZ4_FACTORY = LZ4Factory.fastestInstance();
+
+  /**
+   * Get the unique name of this compression algorithm.
+   * WARNING - Do not change the algorithm name.  See Compression interface for detail.
+   * @return Name of this algorithm.
+   */
+  @Override
+  public String getAlgorithmName() { return "LZ4"; }
```

Comment:
nit: make the algorithm name a public static final string?

Reply:
It can't be static since it's part of the interface.  If I make it final, it becomes non-inheritable.  Since there is no setAlgortihmName, it's okay to leave it as is.